src/main/java/pl/touk/flink/ignite/dialect/IgniteDialect.java

```diff
@@ -31,9 +30,72 @@ public String getLimitClause(long l) {
         return "LIMIT " + l;
     }
 
+    @Override
+    public String quoteIdentifier(String identifier) {
+        return "\"" + identifier + "\"";
+    }
+
     @Override
     public Optional<String> defaultDriverName() {
         return Optional.of("org.apache.ignite.IgniteJdbcThinDriver");
     }
 
+    // WARNING code below was taken from PostgresDialect, it can turn out to be wrong
+
+    private static final int MAX_TIMESTAMP_PRECISION = 6;
+    private static final int MIN_TIMESTAMP_PRECISION = 1;
+
+    private static final int MAX_DECIMAL_PRECISION = 1000;
+    private static final int MIN_DECIMAL_PRECISION = 1;
+
+    @Override
+    public Optional<String> getUpsertStatement(
+            String tableName, String[] fieldNames, String[] uniqueKeyFields) {
+        String uniqueColumns =
+                Arrays.stream(uniqueKeyFields)
+                        .map(this::quoteIdentifier)
+                        .collect(Collectors.joining(", "));
+        String updateClause =
+                Arrays.stream(fieldNames)
+                        .map(f -> quoteIdentifier(f) + "=EXCLUDED." + quoteIdentifier(f))
+                        .collect(Collectors.joining(", "));
+        return Optional.of(
+                getInsertIntoStatement(tableName, fieldNames)
+                        + " ON CONFLICT ("
```

Comment:
I am afraid it won't work. Such syntax seems to be unsupported, according to https://ignite.apache.org/docs/latest/sql-reference/dml. What about using MERGE https://ignite.apache.org/docs/latest/sql-reference/dml#merge and writing some tests? Or can we just signal that UPSERT is not supported and return an empty optional as we do now?